pyriemann/embedding.py

```diff
@@ -96,9 +102,218 @@ def fit_transform(self, X, y=None):
+
+    Attributes
+    ----------
+    embedding_ : ndarray, shape (n_samples, n_components)
+        Stores the embedding vectors
+    error_ : float
+        Reconstruction error associated with `embedding_`
+    data_ : int
+        Training data.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    References
+    ----------
+    .. [1] A. Goh and R. Vidal, "Clustering and dimensionality reduction
+        on Riemannian manifolds", 2008 IEEE Conference on Computer Vision
+        and Pattern Recognition, June 2008
+    """
+
+    def __init__(self, n_components=2, n_neighbors=5,
+                 n_jobs=1, reg=1e-3, **kwargs):
+        self.n_components = n_components
+        self.n_neighbors = n_neighbors
+        self.reg = reg
+        self.null_space_args = kwargs
+
+    def fit(self, X, y=None):
+        """Fit the model from data in X.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            SPD matrices.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+
+        """
+        self.data_ = X
+        self.embedding_, self.error_ = riemann_lle(X,
+                                                   self.n_components,
+                                                   self.n_neighbors,
+                                                   self.reg,
+                                                   self.null_space_args)
+        return self
+
+    def transform(self, X, y=None):
+        pairwise_distances = pairwise_distance(X, self.data_, metric='riemann')
+        ind = np.array([np.argsort(dist)[1:self.n_neighbors + 1]
+                        for dist in pairwise_distances])
+
+        weights = barycenter_weights(X, self.data_, ind, reg=self.reg)
+
+        X_new = np.empty((X.shape[0], self.n_components))
+        for i in range(X.shape[0]):
+            X_new[i] = np.dot(self.embedding_[ind[i]].T, weights[i])
+        return X_new
+
+    def fit_transform(self, X, y=None):
+        self.data_ = X
+        self.embedding_, self.error_ = riemann_lle(X,
+                                                   self.n_components,
+                                                   self.n_neighbors,
+                                                   self.reg,
+                                                   self.null_space_args)
+        return self.embedding_
+
+
+def barycenter_weights(X, Y, indices, reg=1e-3):
+    """Compute Riemannian barycenter weights of X from Y along the first axis.
+    We estimate the weights to assign to each point in Y[indices] to recover
+    the point X[i] by geodesic interpolation. The barycenter weights sum to 1.
+
+    Parameters
+    ----------
+    X : ndarray, shape (n_matrices, n_dim)
+    Y : ndarray, shape (n_matrices, n_dim)
+    indices : ndarray, shape (n_matrices, n_dim)
+        Indices of the points in Y used to compute the barycenter
```

Comment:
why not passing Y[indices] directly to the function?

Reply:
to make it more readable and universal I guess. mostly because they did it the same way in sklearn. can be changed, though.

Reply:
what do you think? is there a computational reason or a reason elsewhere?

Reply:
Uhm, good question. I think not. But I'll check more thoroughly as I go through it again